src/Data/Repositories/WalletRepository.cs

```diff
@@ -134,18 +136,54 @@ public async Task<(bool, string?)> AddRangeAsync(List<Wallet> type)
             return _repository.RemoveRange(types, applicationDbContext);
         }
 
-        public (bool, string?) Update(Wallet type, bool includeKeysUpdate = false)
-        {
+        public async Task<(bool, string?)> Update(Wallet type)
+        { 
             using var applicationDbContext = _dbContextFactory.CreateDbContext();
+            var wallet = await applicationDbContext.Wallets.Include(w => w.Keys).FirstOrDefaultAsync(x => x.Id == type.Id);
 
             type.SetUpdateDatetime();
-            if (!includeKeysUpdate)
-                type.Keys?.Clear();
```

Comment:
Watchout, this is something that was added to avoid conflicts with EF Core, what's the purpose of this removal?

Reply:
because we are now tracking the object instead of getting it from the argument, there is no conflict anymore. We tested it with all cases and it was not needed